// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message TfLiteImageInferenceCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional TfLiteImageInferenceCalculatorOptions ext = 233867214;
  }

  // Path to the TF Lite model (ex: /path/to/modelname.tflite).
  optional string model_path = 1;
  optional bool add_batch_dim = 2 [default = false];
  optional int32 cpu_num_thread = 3 [default = -1];
}
